### PR TITLE
[FIX] Bugs

### PR DESCRIPTION
**Pull request by max3903**

### CODE REVIEW

**Comment by max3903** [comment]
Ping @osi-scampbell

**Review by osi-scampbell — APPROVED** [review]

File: fieldservice_agreement/views/agreement_view.xml [hunk 1]
```diff
@@ -45,4 +44,34 @@
...
+        <field name="name">agreement.search.fields</field>
+        <field name="model">agreement</field>
+        <field name="type">search</field>
+        <field name="inherit_id" ref="agreement.partner_agreement_search_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="/search/filter[1]" position="before">
+                <field name="name"/>
+                <field name="partner_id"/>
+                <field name="agreement_type_id"/>
+                <field name="agreement_subtype_id"/>
```
Comment (max3903) [inline fieldservice_agreement/views/agreement_view.xml]: Those 4 fields are not related to fieldservice. They should be in the original view in the agreement module.

File: fieldservice_agreement/views/agreement_view.xml [hunk 2]
```diff
@@ -45,4 +44,34 @@
...
+    <record id="agreement_end_date_kanban" model="ir.ui.view">
+        <field name="name">agreement.end.date.kanban</field>
+        <field name="model">agreement</field>
+        <field name="inherit_id" ref="agreement.view_project_agreement_kanban"/>
+        <field name="arch" type="xml">
+            <field name="start_date" position="replace">
+                <field name="end_date"/>
+            </field>
+            <xpath expr="//div[@class='o_kanban_record_subtitle text-muted']/t" position="replace">
+                <t t-if="record.end_date.raw_value and record.end_date.raw_value lt (new Date())" t-set="red">oe_kanban_text_red</t>
```
Comment (max3903) [inline fieldservice_agreement/views/agreement_view.xml]: Same here. Nothing related to fieldservice. Should be moved to the agreement module.

File: fieldservice/models/fsm_location.py [hunk 3]
```diff
@@ -64,6 +64,22 @@ def _tz_get(self):
                                      compute='_compute_equipment_ids')
     sublocation_count = fields.Integer(string='Sub Locations',
                                        compute='_compute_sublocation_ids')
+    location_heirarchy = fields.Char(string='Location Heirarchy',
```
Comment (max3903) [inline fieldservice/models/fsm_location.py]: `location_hierarchy`
`Location Hierarchy`

File: fieldservice/models/fsm_location.py [hunk 4]
```diff
@@ -64,6 +64,22 @@ def _tz_get(self):
...
+            parents = self.fsm_parent_id.compute_loc(parents)
+        return parents
+
+    def _compute_location_heirarchy(self):
+        parents = ""
+        for loc in self:
+            parents = loc.compute_loc(parents)
+            parents = parents + str(loc.name)
+            loc.location_heirarchy = parents
+            parents = ""
```
Comment (max3903) [inline fieldservice/models/fsm_location.py]: I think you try to do something like this:
https://github.com/odoo/odoo/blob/11.0/addons/product/models/product.py#L36

File: fieldservice_agreement/views/agreement_view.xml (hunk 1 above)
Comment (smangukiya) [reply fieldservice_agreement/views/agreement_view.xml]: @max3903 : Change has been made here: https://github.com/OCA/contract/pull/279

File: fieldservice_agreement/views/agreement_view.xml (hunk 2 above)
Comment (smangukiya) [reply fieldservice_agreement/views/agreement_view.xml]: @max3903 : Change has been made here: https://github.com/OCA/contract/pull/279

File: fieldservice/models/fsm_location.py (hunk 4 above)
Comment (smangukiya) [reply fieldservice/models/fsm_location.py]: @max3903 : This change has been push to current PR only.

**Comment by smangukiya** [comment]
@max3903 : Please review it again now. Thanks
